### PR TITLE
ref(remove ReactInstanceHolder from AudioModule)

### PR DESCRIPTION
**Pull request by filiprejmus**
These refactoring changes are necessary for the React Native SDK.
Other changes that are needed for the SDK and could be handled in this MR:

-  Substituting ReactInstanceHandler with e.g. Broadcasting in ConnectionService.java
-  Refactoring of DropboxModule.java
-  Making BuildConfig not null when in React Native in AppInfoModule.java

### CODE REVIEW

**Comment by saghul** [comment]
@tmoldovan8x8 Can you PTAL?

File: android/sdk/src/main/java/org/jitsi/meet/sdk/AudioDeviceHandlerConnectionService.java [hunk 1]
```diff
@@ -152,7 +157,7 @@ public void start(AudioModeModule audioModeModule) {
 
     @Override
     public void stop() {
```
Comment (tmoldovan8x8) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/AudioDeviceHandlerConnectionService.java:159]: I guess it also makes sense to set the context to null here, just to make sure there is leak.

File: android/sdk/src/main/java/org/jitsi/meet/sdk/AudioDeviceHandlerConnectionService.java [hunk 2]
```diff
@@ -141,8 +145,9 @@ public void start(AudioModeModule audioModeModule) {
         JitsiMeetLogger.i("Using " + TAG + " as the audio device handler");
 
         module = audioModeModule;
+        context = module.getContext();
```
Comment (tmoldovan8x8) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/AudioDeviceHandlerConnectionService.java:148]: Actually, wouldn't it make more sense to just keep a reference to RNConnectionService? (Again, making sure we set it to null onStop)

**Review by tmoldovan8x8 — CHANGES_REQUESTED** [review]

File: android/sdk/src/main/java/org/jitsi/meet/sdk/AudioDeviceHandlerConnectionService.java (hunk 2 above)
Comment (filiprejmus) [reply android/sdk/src/main/java/org/jitsi/meet/sdk/AudioDeviceHandlerConnectionService.java]: Changed it

**Comment by saghul** [comment]
@tmoldovan8x8 is this ready now?

**Review by tmoldovan8x8 — APPROVED** [review]

**Pull request merged**